src/squidpy/tl/_sliding_window.py

```diff
@@ -0,0 +1,134 @@
+from __future__ import annotations
+
+import math
+from functools import reduce
+from typing import TYPE_CHECKING, Any, Literal, NamedTuple, Optional, Union
+
+import numpy as np
+import pandas as pd
+from anndata import AnnData
+from scanpy import logging as logg
+from spatialdata import SpatialData
+
+from squidpy._docs import d
+from squidpy._utils import NDArrayA
+from squidpy.gr._utils import _save_data
+
+__all__ = ["sliding_window"]
+
+
+@d.dedent
+def sliding_window(
+    adata: AnnData | SpatialData,
+    library_key: str | None = None,
+    window_size: int = 2500,
+    overlap: int = 0,
+    coord_columns: tuple[str, str] = ("globalX", "globalY"),
+    sliding_window_key: str = "sliding_window",
+    spatial_key: str = "spatial",
+    copy: bool = False,
+) -> pd.DataFrame | None:
+    """
+    Divide a tissue slice into regulary shaped spatially contiguous regions (windows).
+
+    Parameters
+    ----------
+    %(adata)s
+    window_size: int
+        Size of the sliding window.
+    %(library_key)s
+    coord_columns: Tuple[str, str]
+        Tuple of column names in `adata.obs` that specify the coordinates (x, y), e.i. ('globalX', 'globalY')
+    sliding_window_key: str
+        Base name for sliding window columns.
+    overlap: int
+        Overlap size between consecutive windows. (0 = no overlap)
+    %(spatial_key)s
+    copy: bool
+        Whether to return a copy of the AnnData object.
+
+    Returns
+    -------
+    If ``copy = True``, returns the sliding window annotation(s) as pandas dataframe
+    Otherwise, stores the sliding window annotation(s) in .obs.
+    """
+    if window_size <= 0:
+        raise ValueError("Window size must be larger than 0.")
+    if overlap < 0:
+        raise ValueError("Overlap must be non-negative.")
+
+    if isinstance(adata, SpatialData):
+        adata = adata.table
+
+    x, y = coord_columns
+
+    if x not in adata.obs or y not in adata.obs:
+        adata.obs[x] = adata.obsm[spatial_key][:, 0]
```

Comment:
Is it really necessary to copy coordinates as new columns to .obs? Should be possible to work directly with .obsm

Reply:
I thought this would be the easiest way to ensure that the user could indicate different x or y variables and after running the sliding window have the x and y coordinates that were used available as reference.